src/unix/fsevents.c

```diff
@@ -267,12 +271,6 @@ static void uv__fsevents_event_cb(ConstFSEventStreamRef streamRef,
         }
       }
 
-#ifdef MAC_OS_X_VERSION_10_7
-      /* Ignore events with path equal to directory itself */
-      if (len == 0)
```

Comment:
I think the easy way to do it is to pass extra argument to `uv__fsevents_init` below, using `stat` result (in `if` statement). `uv__fsevents_init` could just strip the filename from the `realpath` if that extra argument is `1`.

Though, I would suggest you to add an `enum` for this extra argument with values like `kUVFSEventsFile` and `kUVFSEventsDirectory`.